Files/ViewModels/ItemViewModel.cs

```diff
@@ -740,17 +744,16 @@ public async Task LoadExtendedItemProperties(ListedItem item, uint thumbnailSize
                             matchingStorageFile = await GetFileFromPathAsync(item.ItemPath);
                             if (matchingStorageFile != null)
                             {
-                                if (!item.LoadFileIcon) // Loading icon from fulltrust process failed
+                                if (fileIconInfo.IconData == null) // Loading icon from fulltrust process failed
                                 {
                                     using var Thumbnail = await matchingStorageFile.GetThumbnailAsync(ThumbnailMode.SingleItem, thumbnailSize, ThumbnailOptions.UseCurrentScale);
                                     using var headerThumbnail = loadGroupHeaderInfo && isFileTypeGroupMode ? await matchingStorageFile.GetThumbnailAsync(ThumbnailMode.DocumentsView, 36, ThumbnailOptions.UseCurrentScale) : null;
                                     if (Thumbnail != null)
                                     {
                                         await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () =>
                                         {
-                                            item.FileImage = new BitmapImage();
                                             item.CustomIconData = await Thumbnail.ToByteArrayAsync();
-                                            await item.FileImage.SetSourceAsync(Thumbnail);
+                                            item.FileImage = await item.CustomIconData.ToBitmapAsync();
```

Comment:
If there isn't any IconData, then why reference it here?

Reply:
Here `fileIconInfo.IconData` is null (the icon could not be loaded from FTP) but `item.CustomIconData` (icon from StorageFile's thumbnail) is not.

The previous code did not work:
```cs
item.CustomIconData = await Thumbnail.ToByteArrayAsync(); --> this reads the Thumbnail stream to the end
await item.FileImage.SetSourceAsync(Thumbnail);  --> this is left with a stream already read to the end (empty)
```

Reply:
👍